test/cloudant.test.js

```diff
@@ -197,6 +200,93 @@ describe('cloudant connector', function() {
       });
   });
 
+  describe('model with array props gets updated properly', function() {
+    var prod1, prod2;
+    before('create Product', function(done) {
+      Product.create({
+        id: 1,
+        name: 'bread',
+        price: 100,
+        releases: [1, 2, 3],
+        type: ['plain', 'sesame', 'whole wheat'],
+        foo: [{id: 1, name: 'bread1'}, {id: 2, name: 'bread2'}],
+      }, function(err, product) {
+        if (err) return done(err);
+        prod1 = product;
+        Product.create({
+          id: 2,
+          name: 'bagel',
+          price: 100,
+          releases: [1, 2, 3],
+          type: ['plain', 'sesame', 'whole wheat'],
+          foo: [{id: 1, name: 'bagel1'}, {id: 2, name: 'bagel2'}],
+        }, function(err, product) {
+          if (err) return done(err);
+          prod2 = product;
+          done();
+        });
+      });
+    });
+
+    after(function(done) {
+      Product.destroyAll(function(err) {
+        if (err) return done(err);
+        done();
+      });
+    });
+
+    it('updates a single model with array props',
+      function(done) {
+        prod1.setAttribute('type', ['cinnamon raisin']);
+        prod1.setAttribute('releases', [4, 5, 6]);
+        prod1.setAttribute('foo', [{id: 3, name: 'bread3'}]);
+        Product.updateAll({id: 1}, prod1, {}, function(err, res) {
+          if (err) return done(err);
```

Comment:
@jannyHou I did read your comment as it was in my PR and agree that it does the same. However, it is not consistent in this PR. 2 lines down it uses `should.not.exist(err);`. So we should make them consistent on whichever way we are using. :)